.github/workflows/release-with-changesets.yml

```diff
@@ -0,0 +1,137 @@
+# It does magic only if there is a package.json file in the root of the project
+name: Release - if Node project
+
+on:
+  push:
+    branches:
+      - master
+        # The below lines are not enough to have release supported for these branches
+      - next-spec
+      - next-major
+      - next-major-spec
+      - beta
+      - alpha
+      - next
+
+jobs:
+  test-nodejs:
+    # We just check the message of the first commit as there is always just one commit because we squash into one before merging
+    # "commits" contains an array of objects where one of the properties is the commit "message"
+    # Release workflow will be skipped if release conventional commits are not used
+    if: |
+      startsWith( github.repository, 'asyncapi/' ) &&
+      (startsWith( github.event.commits[0].message , 'fix:' ) ||
+      startsWith( github.event.commits[0].message, 'fix!:' ) ||
+      startsWith( github.event.commits[0].message, 'feat:' ) ||
+      startsWith( github.event.commits[0].message, 'feat!:' ))
```

Comment:
here i also want to discuss this. before answering let me state the flow of changesets. 

1. User create the md file stating the package name and level of change.  and pushes the code. 
2. Now when the ci will run and it will reach to this step:
```js
uses: changesets/action@v1
        id: release
        with:
          version: npx -p @changesets/cli changeset version
          commit: version packages
          title: "feat: version packages"
          publish: npx -p @changesets/cli changeset publish
          setupGitUser: false
```
3. Here the changeset action will go through the md files in .changeset folder and based on that will create the pr. with updated version of package. 
for example in apps/generator
```js 
  "name": "@asyncapi/generator",
  "version": "2.0.1",
```

this will be 

```js
  "name": "@asyncapi/generator",
  "version": "3.0.1",
```
Also this pr will also contain the changelogs that we have listed in the readmd file. (i noticed you were searching in the video you shared there the you need to check the pr that was named chore: version packages.) for example check this pr. https://github.com/ayushnau/parser-js/pull/19

4. Now as the pr is merged (will be done by bot in our case) again the ci will run and once again as the ci reaches to the same changeset action step. it will figure out as the pr is created by it only so this time the publish command will be run and the release will happen.

Reply:
ok, found production example -> https://github.com/asyncapi/studio/pull/1112

what I don't understand, why it was triggered by https://github.com/asyncapi/studio/actions/runs/9447823168 if the  PR that triggered this run did not have a changeset file https://github.com/asyncapi/studio/pull/1106/

> Now when the ci will run and it will reach to this step

ok, but this will happen only if conventional commits are respected, fix or feat - which is good

I think mist important change is that no longer a PR with feature triggers release, but a PR created as a result of feature PR - right?

Please provide clear description of that process in readme

Reply:
https://github.com/asyncapi/studio/actions/runs/9447823168 here I believe (though i didnt understand the complete context) what happening is there was other commits that have the md files like here: https://github.com/asyncapi/studio/commit/2c0ab6cdf18630aebf50d21b933fa0d5d3a16b81 when this ran i think the md that is deleted for that file change might have included already in the previous prs but now as the md is deleted now I believe this is the way for changeset to deal with such problem "what if the md file is deleted later on and the release pr is not released yet" so what it does is it check for previous release and check again after that how many md files are created. and that is how it is figuring out total number of md files.

Reply:
now for readme you mean to add the release process is that right.

Reply:
also i checked the documentation there they are suggesting to use the changeset cli to create the md files. 
the documentation whose link https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md

this link needs to be added to the documentation. 

here is how it is working with the use of cli.
[Screencast from 23-07-24 09:49:52 AM IST.webm](https://github.com/user-attachments/assets/b13c43f0-e914-4b95-acca-bbbd0cc3c37a)

Reply:
please comment on this ???

Reply:
we do not need to add dependency, just add instruction how to trigger it with npx
and yeah, if people wanna manual, let them do manual - but instructions lets have CLI examples